react-backend/routes/getAllUsers.js

```diff
@@ -1,15 +1,24 @@
 const express = require('express');
 
+const jwt = require('jsonwebtoken');
+
 const router = express.Router();
 
 const getAllUsers = require('../database/queries/getAllUsers');
 
 router.get('/', (req, res) => {
-  getAllUsers()
-    .then(users => res.send({ users }))
-    .catch(err => {
+  jwt.verify(req.headers.authorization, process.env.SECRET, (err, payload) => {
+    if (err) {
       console.log(err);
-    });
+      res.send(JSON.stringify({ error: 'Not logged in' }));
+    } else {
+      getAllUsers()
+        .then(users => res.send({ users }))
+        .catch((err) => {
+          console.log(err);
```

Comment:
in this error case we need to send a 500 error to the front end.

Reply:
here the same: res.status(500).send(err)